### PR TITLE
chore(sql): improve strpos function

### PR DESCRIPTION
**Pull request by puzpuzpuz**
A follow-up for #1506

Includes the following:
* Introduces const variations for strpos functions since it's going to be a popular case for users
* Adds missing checks for zero char into `StrPosCharFunctionFactory`
* Adds tests to cover new code
* Updates the strpos benchmark

Results on my local (Ubuntu 20.04, OpenJDK 11.0.11):
```
Benchmark                              Mode  Cnt    Score     Error  Units
StrPosBenchmark.testBaseline           avgt    3   11.539 ±   0.596  ns/op
StrPosBenchmark.testCharConstOverload  avgt    3  504.146 ± 230.001  ns/op
StrPosBenchmark.testCharOverload       avgt    3  509.196 ±  62.726  ns/op
StrPosBenchmark.testStrConstOverload   avgt    3  567.364 ±  60.399  ns/op
StrPosBenchmark.testStrOverload        avgt    3  540.781 ±  75.134  ns/op
```

P.S. The results are more or less close to each other. What seems weird is the slightly higher error rate in `testCharConstOverload`, but not that it's something critical.

### CODE REVIEW

File: benchmarks/src/main/java/org/questdb/StrPosBenchmark.java [hunk 1]
```diff
@@ -70,10 +70,8 @@ public StrPosBenchmark() {
         strings = new String[N];
         for (int i = 0; i < N; i++) {
             builder.setLength(0);
-            int startLen = rnd.nextInt(1000);
-            for (int j = 0; j < startLen; j++) {
-                builder.append('a');
-            }
+            int startLen = rnd.nextInt(32);
+            builder.append("a".repeat(startLen));
```
Comment (bluestreak01) [inline benchmarks/src/main/java/org/questdb/StrPosBenchmark.java:74]: This is JDK11 only API

I specifically removed it here:

https://github.com/questdb/questdb/pull/1533

There is an open issue regarding jdk8 compatibility 

https://github.com/questdb/questdb/issues/1532

**Review by bluestreak01 — CHANGES_REQUESTED** [review]

File: benchmarks/src/main/java/org/questdb/StrPosBenchmark.java (hunk 1 above)
Comment (puzpuzpuz) [reply benchmarks/src/main/java/org/questdb/StrPosBenchmark.java:74]: My bad. Let me revert this part of the change.

Comment (puzpuzpuz) [reply benchmarks/src/main/java/org/questdb/StrPosBenchmark.java]: Fixed in 9e96fc609fad87981548f79ca193e7d9294f2bbe

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 32 / 35 (91.43%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/engine/functions/str/StrPosFunctionFactory.java|19|21|90.48%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/str/StrPosCharFunctionFactory.java|13|14|92.86%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 32 / 35 (91.43%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/engine/functions/str/StrPosFunctionFactory.java|19|21|90.48%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/str/StrPosCharFunctionFactory.java|13|14|92.86%|

**Comment by bluestreak01** [comment]
could we improve test coverage ?

**Comment by puzpuzpuz** [comment]
> could we improve test coverage ?

The coverage is not 100% mostly due to the empty string literal problem. Once https://github.com/questdb/questdb/blob/master/core/src/test/java/io/questdb/griffin/engine/functions/str/StrPosFunctionFactoryTest.java#L82 gets resolved, I'm going to return to this test and update it. Does it make sense?

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 32 / 35 (91.43%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/engine/functions/str/StrPosFunctionFactory.java|19|21|90.48%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/str/StrPosCharFunctionFactory.java|13|14|92.86%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 32 / 35 (91.43%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/engine/functions/str/StrPosFunctionFactory.java|19|21|90.48%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/str/StrPosCharFunctionFactory.java|13|14|92.86%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

coverage check fail. please retry. :fearful:

[Please let me know](https://github.com/naver/cover-checker/issues/new) when error again.

com.naver.nid.cover.parser.coverage.exception.ParseException: java.io.FileNotFoundException: /home/vsts/work/1/s/core/CCReport43F6D5EF/jacoco.xml (No such file or directory)

**Comment by ideoma** [comment]
#### [PR Coverage check]

coverage check fail. please retry. :fearful:

[Please let me know](https://github.com/naver/cover-checker/issues/new) when error again.

com.naver.nid.cover.parser.coverage.exception.ParseException: java.io.FileNotFoundException: /home/vsts/work/1/s/core/CCReport43F6D5EF/jacoco.xml (No such file or directory)

**Comment by ideoma** [comment]
#### [PR Coverage check]

coverage check fail. please retry. :fearful:

[Please let me know](https://github.com/naver/cover-checker/issues/new) when error again.

com.naver.nid.cover.parser.coverage.exception.ParseException: java.io.FileNotFoundException: /usr/share/vsts-agent/_work/1/s/CCReport43F6D5EF/jacoco.xml (No such file or directory)

**Comment by ideoma** [comment]
#### [PR Coverage check]

coverage check fail. please retry. :fearful:

[Please let me know](https://github.com/naver/cover-checker/issues/new) when error again.

com.naver.nid.cover.parser.coverage.exception.ParseException: java.io.FileNotFoundException: /home/vsts/work/1/s/CCReport43F6D5EF/jacoco.xml (No such file or directory)

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 32 / 35 (91.43%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/engine/functions/str/StrPosFunctionFactory.java|19|21|90.48%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/str/StrPosCharFunctionFactory.java|13|14|92.86%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 32 / 35 (91.43%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/engine/functions/str/StrPosFunctionFactory.java|19|21|90.48%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/str/StrPosCharFunctionFactory.java|13|14|92.86%|

**Review by bluestreak01 — APPROVED** [review]

**Pull request merged**